lib/fs.js

```diff
@@ -211,6 +213,7 @@ fs.access = function(path, mode, callback) {
     throw new TypeError('"callback" argument must be a function');
   }
 
+  path = getPathFromURL(path);
```

Comment:
The WHATWG URL parsing could escape the null byte if it's not in the hostname ('\u0000' -> '%00', e.g. `new URL('file://hostname/a/b/c\u0000')`), thus bypassing the null check, so might worth add a test for it. Is it possible for `getPathFromURL` to put null bytes in a path without one? If not, maybe the `nullCheck` can be performed before `getPathFromURL`.